packages/cli/src/lib/implementation/core-config-options.integration.test.ts

```diff
@@ -1,20 +1,47 @@
+import { join } from 'node:path';
 import { describe, expect } from 'vitest';
 import { CoreConfig } from '@code-pushup/models';
 import { objectToCliArgs } from '@code-pushup/utils';
 import { yargsCli } from '../yargs-cli';
 import { yargsCoreConfigOptionsDefinition } from './core-config-options';
 
 describe('configOptions', () => {
-  function cli(args: Record<string, unknown>): Promise<CoreConfig> {
-    return yargsCli(objectToCliArgs(args), {
-      options: yargsCoreConfigOptionsDefinition(),
-    }).parseAsync() as unknown as Promise<CoreConfig>;
+  function argsFromCli(args: Record<string, unknown>): Promise<CoreConfig> {
+    return yargsCli(
+      objectToCliArgs({
+        ...args,
+        config: join('code-pushup.config.ts'),
+      }),
+      {
+        options: yargsCoreConfigOptionsDefinition(),
+      },
+    ).parseAsync() as unknown as Promise<CoreConfig>;
   }
 
   it('should fill defaults', async () => {
-    const config = await cli({});
+    const config = await argsFromCli({ config: 'code-pushup.config.ts' });
     expect(config).toBeDefined();
     // only _ and $0
-    expect(Object.keys(config)).toHaveLength(2);
+    expect(Object.keys(config)).toHaveLength(3);
   });
+
+  it.each([
+    // defaults
+    [{}, {}],
+    // persist.outputDir
+    [{ 'persist.outputDir': 'my-dir' }, { outputDir: 'my-dir' }],
+    // persist.filename
+    [{ 'persist.filename': 'my-report' }, { filename: 'my-report' }],
+    // persist.format
+    [{ 'persist.format': 'md' }, { format: ['md'] }],
+    [{ 'persist.format': ['md', 'json'] }, { format: ['md', 'json'] }],
+    // [{ 'persist.format': 'md,json' }, { format: ['md', 'json'] }], @TODO comment in when config auto-loading is implemented
+  ])(
+    'should parse persist options %s correctly as %j',
```

Comment:
Since the first parameter is an object, I'd expect the JSON serialization to be more readable than the string serialization (`[object Object]`, I'm guessing).

```suggestion
    'should parse persist options %j correctly as %j',
```

Reply:
I did not know about %s is for string and %j for json :o 

Thx for the info! 

- %p - [pretty-format](https://www.npmjs.com/package/pretty-format).
- %s- String.| %d- Number. | %i - Integer.
- %f - Floating point value.| %j - JSON.| %o - Object.
- %# - Index of the test case.| %% - single percent sign ('%').